packages/data-stores/src/reader/mutations/use-subscription-manager-site-delivery-frequency-mutation.ts

```diff
@@ -0,0 +1,62 @@
+import { useMutation, useQueryClient } from 'react-query';
+import { callApi } from '../helpers';
+import { useIsLoggedIn } from '../hooks';
+import type { SiteSubscriptionDeliveryFrequency } from '../types';
+
+type SiteSubscriptionDeliveryFrequencyParams = {
+	delivery_frequency: SiteSubscriptionDeliveryFrequency;
+	blog_id: number | string;
+};
+
+type SubscriptionResponse = {
+	blog_ID: string;
+	delivery_frequency: SiteSubscriptionDeliveryFrequency;
+	status: string;
+	ts: Date;
+};
+
+type SiteSubscriptionDeliveryFrequencyResponse = {
+	success: boolean;
+	subscribed: boolean;
+	subscription: SubscriptionResponse | null;
+};
+
+const useSubscriptionManagerSiteDeliveryFrequencyMutation = () => {
+	const isLoggedIn = useIsLoggedIn();
+	const queryClient = useQueryClient();
+	return useMutation(
+		async ( params: SiteSubscriptionDeliveryFrequencyParams ) => {
+			if ( ! params.blog_id || ! params.delivery_frequency ) {
+				throw new Error(
+					// reminder: translate this string when we add it to the UI
+					'Something went wrong while saving the delivery frequency.'
```

Comment:
Is there a more helpful error message we could show here and below?

Reply:
I don't know. We don't get much back from the backend in this case (`success: false)`, and when one of the paramaters is missing it's an implementation error and not a user error. We also don't handle the display of errors in the frontend (something I'm planning on picking up at our next design sync).